pkg/rpcserver/rpcserver.go

```diff
@@ -306,25 +327,17 @@ func (serv *server) handleRunnerConn(ctx context.Context, eg *errgroup.Group,
 		return err
 	}
 
-	serv.checkOnce.Do(func() {
-		// Run the machine check.
-		eg.Go(func() error {
-			if err := serv.runCheck(ctx, &info); err != nil {
-				return fmt.Errorf("%w: %w", errFatal, err)
-			}
-			return nil
-		})
-	})
-
-	if serv.triagedCorpus.Load() {
-		eg.Go(runner.SendCorpusTriaged)
+	select {
+	case serv.onHandshake <- &info:
+	default:
 	}
 
-	go func() {
-		<-ctx.Done()
-		runner.Stop()
-	}()
-	return serv.connectionLoop(runner)
+	eg, ctx := errgroup.WithContext(baseCtx)
+	eg.Go(runner.SendCorpusTriaged)
```

Comment:
We should do this only when serv.triagedCorpus is set, and we can do this synchronously w/o the additional errgroup to simplify code.